src/status_im/transport/inbox.cljs

```diff
@@ -219,18 +221,22 @@
  :inbox/request-messages
  (fn [{:keys [db now]} [_ {:keys [from topics discover?]}]]
    (let [web3       (:web3 db)
+         account    (:account/account db)
          wnode      (get-current-wnode-address db)
-         topics     (or topics
-                        (map #(:topic %) (vals (:transport/chats db))))
-         from       (or from (:inbox/last-request db) nil)
-         sym-key-id (:inbox/sym-key-id db)]
-     {::request-messages {:wnode      wnode
-                          :topics     (if discover?
-                                        (conj topics (transport.utils/get-topic constants/contact-discovery))
-                                        topics)
-                          ;;TODO (yenda) fix from, right now mailserver is dropping us
-                          ;;when we send a requestMessage with a from field
-                          ;;:from       from
-                          :sym-key-id sym-key-id
-                          :web3       web3}
-      :db (assoc db :inbox/last-request (quot now 1000))})))
+         topics     (cond-> (or topics
+                                (map #(:topic %) (vals (:transport/chats db))))
+                      discover? (conj (transport.utils/get-topic constants/contact-discovery)))
+         from       (or from (:last-request account))
```

Comment:
Seems like we use a single cursor for different topics:

`last-request` is 20
I ask for topic `B` with from = 20, now `last-request` is 30
Next time I query the mailserver, I ask for all the topics with from = 30.

Seems to me we should ask from 20 the second time as well, otherwise we might lose some messages.

This makes the system quite difficult to reason about.

Two ways to handle this I can see, either we have a cursor for each topic + discovery (something like `transport/chats chat-id last-request`), or we always fetch all the topics. What do you think?

Reply:
> Two ways to handle this I can see, either we have a cursor for each topic + discovery (something like transport/chats chat-id last-request), or we always fetch all the topics. What do you think?

@yenda what do you think regarding the above question?

Reply:
there's also the option as mentioned to don't update the cursor when you make one-off requests (specify a topic), only when we ask for all the topics we update it.